client_app/src/components/shelter/ShiftDetailsTable.js

```diff
@@ -93,32 +106,52 @@ export const ShiftDetailsTable = props => {
     }
   }
 
+  const renderCondensedCols = () => {
+    return (shiftDetailsData.data.map((shift) => (
+      <tr key={shift.id} style={{background:getShiftColor(shift)}}>
+        <td>{format(shift.startTime, "MM/dd/yyyy")}</td>
+        <td>{shift.name ? shift.name + ": " + format(shift.startTime, "hh:mm aaaaa'm'") + ' - ' + format(shift.endTime, "hh:mm aaaaa'm'") : format(shift.startTime, "hh:mm aaaaa'm'") + ' - ' + format(shift.endTime, "hh:mm aaaaa'm'")}</td>
+        <td>{renderCoverage(shift)}</td>
+      </tr>)))
+  }
+
+  const renderAllCols = () => {
+    return (shiftDetailsData.data.map((shift) => (
+      <tr key={shift.id} style={{background:getShiftColor(shift)}}>
+        <td>{format(shift.startTime, "MM/dd/yyyy")}</td>
+        <td>{shift.name ? shift.name + ": " + format(shift.startTime, "hh:mm aaaaa'm'") + ' - ' + format(shift.endTime, "hh:mm aaaaa'm'") : format(shift.startTime, "hh:mm aaaaa'm'") + ' - ' + format(shift.endTime, "hh:mm aaaaa'm'")}</td>
+        <td>{renderCoverage(shift)}</td>
+        <td>{viewRosterButton()}</td>
+        <td>{shift.status ? openStatusButton(shift) : closedStatusButton(shift)}</td>
+        <td>{editButton(shift)}</td>
+        <td>{emailButton()}</td>
+      </tr>
+    )))
+  }
+
+  const getShiftColor = shift => {
+    if (format(shift.startTime, "MM/dd/yyyy") != prevStartDate) {
+      prevStartDate = format(shift.startTime, "MM/dd/yyyy");
+      return "#e6e6e6";
+    } else {
+      return "white";
+    }
+  }
+
   return (
     <div className="shiftdetails-container">
+      {(!areColsExpanded && isMobileView) && <button onClick={() => {setAreColsExpanded(true)}}>View All Columns</button>}
+      {(isMobileView && areColsExpanded) && <button onClick={() => {setAreColsExpanded(false)}}>Collapse Columns</button>}
       <table className="shiftsTable">
         <thead className="shiftsTableHeader" style={{position:stickyHeader,zIndex:headerZIndex,top:-1}}>
```

Comment:
I like the collapsable columns, its easy to view everything on the phone set up

Reply:
She liked it was easy to see